x-pack/plugins/index_management/public/application/lib/data_streams.tsx

```diff
@@ -38,3 +39,17 @@ export const isSelectedDataStreamHidden = (
       ?.hidden
   );
 };
+
+export const getLifecycleValue = (lifecycle: DataStream['lifecycle']) => {
+  if (!lifecycle?.enabled) {
+    return i18n.translate('xpack.idxMgmt.dataStreamList.dataRetentionDisabled', {
+      defaultMessage: 'Disabled',
+    });
+  } else if (!lifecycle?.data_retention) {
+    return i18n.translate('xpack.idxMgmt.dataStreamList.dataRetentionInfinite', {
+      defaultMessage: 'Keep data indefinitely',
```

Comment:
This is more polish, but I wonder if it'd be better to use the `infinity` [icon](https://eui.elastic.co/#/display/icons) with a tooltip or something in the table.

Reply:
I've changed this so that we can show the icon in the table and keep it more compact, but still showing it as plain text in the flyout to be more explicit.

![Screenshot 2023-10-13 at 09 53 33](https://github.com/elastic/kibana/assets/1191206/17050c46-3d53-4d06-89bd-cab46708e00b)